core/station/impl/src/services/user.rs

```diff
@@ -110,13 +110,26 @@ impl UserService {
     ///
     /// This method should only be called by a system call (self canister call or controller).
     pub fn add_user(&self, input: AddUserOperationInput) -> ServiceResult<User> {
+        self.add_user_with_id(input, None)
+    }
+
+    /// Creates a new user with the given user details and returns the created user.
+    ///
+    /// This method should only be called by a system call (self canister call or controller).
+    pub fn add_user_with_id(
+        &self,
+        input: AddUserOperationInput,
+        with_id: Option<UserId>,
+    ) -> ServiceResult<User> {
         for identity in input.identities.iter() {
             self.assert_identity_has_no_associated_user(identity, None)?;
         }
 
+        let user_id = with_id.unwrap_or_else(|| *Uuid::new_v4().as_bytes());
```

Comment:
Shouldn't we also check if this user ID already exists (just like for users groups etc.)?

Reply:
Good catch